server/services/deliusOfficeLocationFilter.test.ts

```diff
@@ -0,0 +1,99 @@
+import ReferenceDataService from './referenceDataService'
+import DeliusOfficeLocationFilter from './deliusOfficeLocationFilter'
+import InterventionsService from './interventionsService'
+import apiConfig from '../config'
+import MockReferenceDataService from '../routes/testutils/mocks/mockReferenceDataService'
+import interventionFactory from '../../testutils/factories/intervention'
+import deliusOfficeLocationFactory from '../../testutils/factories/deliusOfficeLocation'
+
+jest.mock('./interventionsService')
+jest.mock('./referenceDataService')
+
+describe(DeliusOfficeLocationFilter, () => {
+  const interventionsService = new InterventionsService(
+    apiConfig.apis.interventionsService
+  ) as jest.Mocked<InterventionsService>
+  const referenceDataService = new MockReferenceDataService() as jest.Mocked<ReferenceDataService>
+  const deliusOfficeLocationProvider = new DeliusOfficeLocationFilter(referenceDataService, interventionsService)
+  describe('with an intervention having an nps region', () => {
+    it('should return an office within the same nps region', async () => {
+      const intervention = interventionFactory.build({
+        npsRegion: { id: 'B', name: 'North West' },
+        pccRegions: [],
+      })
+      const deliusOfficeLocation = deliusOfficeLocationFactory.build({ probationRegionId: 'B' })
+      interventionsService.getIntervention.mockResolvedValue(intervention)
+      referenceDataService.getProbationOffices.mockResolvedValue([deliusOfficeLocation])
+      const office = await deliusOfficeLocationProvider.findOfficesByInterventionId('', '')
+      expect(office.length).toEqual(1)
+      expect(office[0].probationRegionId).toEqual('B')
+    })
+    it('should filter out offices with empty crs location ids', async () => {
+      const intervention = interventionFactory.build({
+        npsRegion: { id: 'B', name: 'North West' },
+        pccRegions: [],
+      })
+      const emptyOfficeLocation = deliusOfficeLocationFactory.build({
+        deliusCRSLocationId: '',
+      })
+      interventionsService.getIntervention.mockResolvedValue(intervention)
+      referenceDataService.getProbationOffices.mockResolvedValue([emptyOfficeLocation])
+      const office = await deliusOfficeLocationProvider.findOfficesByInterventionId('', '')
+      expect(office.length).toEqual(0)
+    })
+  })
+  describe('with an intervention having an pcc region', () => {
+    it('should return an office within the same pcc region', async () => {
```

Comment:
I don't understand the behaviour that's being described here - as far as I can tell, the offices do not have a PCC region associated with them. Should this be "an office within the same NPS region"?